boards/arm/mimxrt685_evk/pinmux.c

```diff
@@ -192,6 +192,92 @@ static int mimxrt685_evk_pinmux_init(struct device *dev)
 	IOPCTL_PinMuxSet(IOPCTL, 1U, 5U, port1_pin5_config);
 #endif
 
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm5), okay) && CONFIG_SPI
+	const uint32_t port1_pin3_config = (/* Pin is configured as FC5_SCK */
+		IOPCTL_PIO_FUNC1 |
+		/* Disable pull-up / pull-down function */
+		IOPCTL_PIO_PUPD_DI |
+		/* Enable pull-down function */
+		IOPCTL_PIO_PULLDOWN_EN |
+		/* Enables input buffer function */
+		IOPCTL_PIO_INBUF_EN |
+		/* Normal mode */
+		IOPCTL_PIO_SLEW_RATE_NORMAL |
+		/* Normal drive */
+		IOPCTL_PIO_FULLDRIVE_DI |
+		/* Analog mux is disabled */
+		IOPCTL_PIO_ANAMUX_DI |
+		/* Pseudo Output Drain is disabled */
+		IOPCTL_PIO_PSEDRAIN_DI |
+		/* Input function is not inverted */
+		IOPCTL_PIO_INV_DI);
+	/* PORT1 PIN3 (coords: G16) is configured as FC5_SCK */
+	IOPCTL_PinMuxSet(IOPCTL, 1U, 3U, port1_pin3_config);
+
+	const uint32_t port1_pin4_config = (/* Pin is configured as FC5_MISO */
+		IOPCTL_PIO_FUNC1 |
+		/* Disable pull-up / pull-down function */
+		IOPCTL_PIO_PUPD_DI |
+		/* Enable pull-down function */
+		IOPCTL_PIO_PULLDOWN_EN |
+		/* Enables input buffer function */
+		IOPCTL_PIO_INBUF_EN |
+		/* Normal mode */
+		IOPCTL_PIO_SLEW_RATE_NORMAL |
+		/* Normal drive */
+		IOPCTL_PIO_FULLDRIVE_DI |
+		/* Analog mux is disabled */
+		IOPCTL_PIO_ANAMUX_DI |
+		/* Pseudo Output Drain is disabled */
+		IOPCTL_PIO_PSEDRAIN_DI |
+		/* Input function is not inverted */
+		IOPCTL_PIO_INV_DI);
+	/* PORT1 PIN4 (coords: G17) is configured as FC5_TXD_SCL_MISO_WS */
+	IOPCTL_PinMuxSet(IOPCTL, 1U, 4U, port1_pin4_config);
+
+	const uint32_t port1_pin5_config = (/* Pin is configured as FC5_MOSI */
+		IOPCTL_PIO_FUNC1 |
+		/* Disable pull-up / pull-down function */
+		IOPCTL_PIO_PUPD_DI |
+		/* Enable pull-down function */
+		IOPCTL_PIO_PULLDOWN_EN |
+		/* Enables input buffer function */
+		IOPCTL_PIO_INBUF_EN |
+		/* Normal mode */
+		IOPCTL_PIO_SLEW_RATE_NORMAL |
+		/* Normal drive */
+		IOPCTL_PIO_FULLDRIVE_DI |
+		/* Analog mux is disabled */
+		IOPCTL_PIO_ANAMUX_DI |
+		/* Pseudo Output Drain is disabled */
+		IOPCTL_PIO_PSEDRAIN_DI |
+		/* Input function is not inverted */
+		IOPCTL_PIO_INV_DI);
+	/* PORT1 PIN5 (coords: J16) is configured as FC5_RXD_SDA_MOSI_DATA */
+	IOPCTL_PinMuxSet(IOPCTL, 1U, 5U, port1_pin5_config);
+
+	const uint32_t port1_pin6_config = (/* Pin is configured as FC5_SSEL0 */
```

Comment:
If you want the ability to have either Flexcomm's SS or GPIO CS as mentioned in dts file make a change similar to the one below:

https://github.com/zephyrproject-rtos/zephyr/blob/c5d85e175fae85dce4236d20a92717e084b57709/boards/arm/lpcxpresso55s69/pinmux.c#L128-L138

Reply:
Thank you, this is good to know. In this case, the configuration is for FlexComm5 and not HS_LSPI, hence we need not apply this.